### PR TITLE
Add info on requests and limits for functions

### PR DESCRIPTION
**Pull request by kazydek**
**Description**

Changes proposed in this pull request:

- Created the configuration doc for serverless that documents default `buildLimits` and `buildRequests` for CPU and memory and the Minikube overrides.
- Mentioned in docs that Istio is required for Namespaces in which you create functions (sidecar must be injected)
- Mentioned that currently there is no automated validation for CPU and memory requests and limits for functions.
- Clearly stated in docs that the values for CPU and memory requests must be not lower than the default ones, and that values for requests must be lower than those for the limits.
- Stated that LimitRanges you set for resources in a given Namespace for functions must not be lower than the default values.

**Related issue(s)**
See also #8176

### CODE REVIEW

File: docs/serverless/08-01-create-function.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ Follows these steps:
     export NAMESPACE={FUNCTION_NAMESPACE}
     ```
 
+    > **CAUTION:** If you create a new Namespace, do not disable sidecar injection in it as Serverless requires Istio for other resources to communicate with functions correctly. Also, if you apply custom [LimitRanges](https://kyma-project.io/docs/#details-resource-quotas) for a new Namespace, they must be higher than the default values.
```
Comment (magicmatatjahu) [inline docs/serverless/08-01-create-function.md:25]: I think that this Caution should be not in tutorial, but in other `more often read` place like Overview, or Details.

File: docs/serverless/08-01-create-function.md [hunk 2]
```diff
@@ -62,6 +64,8 @@ Follows these steps:
 
 1. Create a Namespace or select one from the drop-down list in the top navigation panel.
 
+    > **CAUTION:** If you create a new Namespace, do not disable sidecar injection in it as Serverless requires Istio for other resources to communicate with functions correctly. Also, if you apply custom [LimitRanges](https://kyma-project.io/docs/#details-resource-quotas) for a new Namespace, they must be higher than the default values.
```
Comment (magicmatatjahu) [inline docs/serverless/08-01-create-function.md:67]: I think that this Caution should be not in tutorial, but in other more often read place like Overview, or Details.

File: docs/serverless/05-01-serverless.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+>**TIP:** To learn more about how to use overrides in Kyma, see the following documents:
+>* [Helm overrides for Kyma installation](/root/kyma/#configuration-helm-overrides-for-kyma-installation)
+>* [Top-level charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-top-level-charts-overrides)
+
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values for both cluster and local installations.
+
+>**CAUTION:** Currently, there is no automated validation for CPU and memory requests and limits. If you update their default values, make sure that the values for CPU and memory requests are not lower than the default ones, and that values for requests are lower than those for the limits.
```
Comment (magicmatatjahu) [inline docs/serverless/05-01-serverless.md:16]: ...and that values for requests are lower or **equal** than those for the limits.

File: docs/serverless/08-01-create-function.md (hunk 2 above)
Comment (kazydek) [reply docs/serverless/08-01-create-function.md:67]: yup, I had the same concern but I guess you are right 👍

File: docs/serverless/08-01-create-function.md (hunk 1 above)
Comment (kazydek) [reply docs/serverless/08-01-create-function.md:25]: yup, I had the same concern but I guess you are right 👍

File: docs/serverless/02-01-serverless.md [hunk 4]
```diff
@@ -6,6 +6,8 @@ Serverless relies on [Knative Serving](https://knative.dev/docs/serving/) for de
 
 ![Serverless architecture](./assets/serverless-architecture.svg)
 
+> **CAUTION:** Serverless imposes some requirements on the setup of Namespaces. If you create a new Namespace, do not disable sidecar injection in it as Serverless requires Istio for other resources to communicate with functions correctly. Also, if you apply custom [LimitRanges](https://kyma-project.io/docs/#details-resource-quotas) for a new Namespace, they must be higher than the default values.
```
Comment (michal-hudy) [inline docs/serverless/02-01-serverless.md:9]: `they must be higher than the default values.` -> `they must be higher or equal than the default values.` and link to default values

File: docs/serverless/05-01-serverless.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+>**TIP:** To learn more about how to use overrides in Kyma, see the following documents:
+>* [Helm overrides for Kyma installation](/root/kyma/#configuration-helm-overrides-for-kyma-installation)
+>* [Top-level charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-top-level-charts-overrides)
+
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values for both cluster and local installations.
+
+>**CAUTION:** Currently, there is no automated validation for CPU and memory requests and limits. If you update their default values, make sure that the values for CPU and memory requests are not lower than the default ones, and that values for requests are lower than or equal to those for the limits.
```
Comment (michal-hudy) [inline docs/serverless/05-01-serverless.md:16]: I'm not sure if you mention building resources or runtime resources? So I'm also not sure what does it mean? `If you update their default values, make sure that the values for CPU and memory requests are not lower than the default ones`

Comment (kazydek) [reply docs/serverless/05-01-serverless.md:16]: I will specify that this concern building resources.

File: docs/serverless/02-01-serverless.md (hunk 4 above)
Comment (aerfio) [reply docs/serverless/02-01-serverless.md:9]: > ... they must be higher than the default values

default values for what -> please specify that we are talking about build  step,resources for build jobs

Comment (kazydek) [reply docs/serverless/02-01-serverless.md:9]: yup, corrected

File: docs/serverless/05-01-serverless.md [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+>* [Helm overrides for Kyma installation](/root/kyma/#configuration-helm-overrides-for-kyma-installation)
+>* [Top-level charts overrides](/root/kyma/#configuration-helm-overrides-for-kyma-installation-top-level-charts-overrides)
+
+## Configurable parameters
+
+This table lists the configurable parameters, their descriptions, and default values for both cluster and local installations.
+
+| Parameter | Description | Default value | Minikube override |
+|-----------|-------------|---------------|---------------|
+| **containers.manager.envs.buildRequestsCPU** | Specifies the number of CPUs requested by the image building Pod to operate. | `700m` | `100m`|
```
Comment (bszwarc) [inline docs/serverless/05-01-serverless.md:18]:
```suggestion
| **containers.manager.envs.buildRequestsCPU** | Number of CPUs requested by the image building Pod to operate. | `700m` | `100m`|
```
also maybe `image-building` Pod - as an adjective?

File: docs/serverless/05-01-serverless.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+| **containers.manager.envs.buildRequestsMemory** | Specifies the amount of memory requested by the image building Pod to operate.  | `700Mi` | `200Mi` |
```
Comment (bszwarc) [inline docs/serverless/05-01-serverless.md:19]:
```suggestion
| **containers.manager.envs.buildRequestsMemory** | Amount of memory requested by the image building Pod to operate.  | `700Mi` | `200Mi` |
```

File: docs/serverless/05-01-serverless.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+| **containers.manager.envs.buildLimitsCPU** | Specifies the maximum number of CPUs available for the image building Pod to use. | `1100m` | `200m` |
```
Comment (bszwarc) [inline docs/serverless/05-01-serverless.md:20]:
```suggestion
| **containers.manager.envs.buildLimitsCPU** | Maximum number of CPUs available for the image building Pod to use. | `1100m` | `200m` |
```

File: docs/serverless/05-01-serverless.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+| **containers.manager.envs.buildLimitsMemory** | Specifies the maximum amount of memory available for the image building Pod to use. | `1100Mi` | `400Mi` |
```
Comment (bszwarc) [inline docs/serverless/05-01-serverless.md:21]:
```suggestion
| **containers.manager.envs.buildLimitsMemory** | Maximum amount of memory available for the image building Pod to use. | `1100Mi` | `400Mi` |
```

File: docs/serverless/05-01-serverless.md (hunk 6 above)
Comment (kazydek) [reply docs/serverless/05-01-serverless.md:18]: yup :)

**Pull request merged**